### PR TITLE
Ranking: replace custom aggregate with jsonb_object_agg

### PR DESCRIPTION
**Pull request by camdencheek**
This replaces our custom aggregation based on `jsonb_object()` with the builtin `jsonb_object_agg()`, which is much more efficient. I think this is because it does not merge each new row into a new JSONB object (`jsonb_object` takes exactly two arguments). 

Current pattern, takes 783ms locally:
```sql
with objects as ( 
  select jsonb_build_object(key, value) as object 
  from (select generate_series(1,6000)::text as key, generate_series(1,6000)::text as value) t1
) 
select sg_jsonb_concat_agg(object) from objects;
```

New pattern, takes 10ms locally:
```sql
with objects as (
  select key, value 
  from (select generate_series(1,6000)::text as key, generate_series(1,6000)::text as value) t1
) 
select jsonb_object_agg(key, value) from objects;
```

## Test plan

Running unit tests and integration tests.

### CODE REVIEW

File: enterprise/internal/codeintel/ranking/internal/store/ranking.go [hunk 1]
```diff
@@ -360,11 +360,12 @@ inserted AS (
 	SELECT
 		temp.repository_id,
 		%s,
-		sg_jsonb_concat_agg(temp.row)
```
Comment (efritz) [inline enterprise/internal/codeintel/ranking/internal/store/ranking.go:363]: This is also used in the ON CONFLICT clause.

File: migrations/frontend/1678832491_remove_sg_jsonb_concat_agg/up.sql
Comment (efritz) [inline migrations/frontend/1678832491_remove_sg_jsonb_concat_agg/up.sql:1]: You'll need to go generate in internal/database

Comment (camdencheek) [reply migrations/frontend/1678832491_remove_sg_jsonb_concat_agg/up.sql:1]: already on it 🙂

Comment (camdencheek) [reply migrations/frontend/1678832491_remove_sg_jsonb_concat_agg/up.sql:1]: I really like this "pair programming over draft PR" flow

**Comment by camdencheek** [comment]
@efritz I'm assuming we want to backport this

**Review by efritz — APPROVED** [review]
Validated this change locally.

**Pull request merged**